### PR TITLE
Use template to generate copy_config_files.sh

### PR DESCRIPTION
**Pull request by ffoulkes**
- Create a Jinja template and Python driver to generate the copy_config_files.sh file. This gives us more flexibility the cmake configure_files() command.

I'm offering this as an alternative to PR https://github.com/ipdk-io/networking-recipe/pull/325

### CODE REVIEW

**Comment by ffoulkes** [comment]
The `genscript.py` utility requires `pip install Jinja2` on the development system to work.

-----

According to usage text the shell script displays:
```text
    echo " - Usage: sudo copy_config_files.sh <P4CP_INSTALL> <SDE_INSTALL>"
```
the user is expected to run the script using `sudo` (or else be running as `root`). This makes perfect sense. The purpose of the script is to update system directories. It inherently requires administrative privilege. It's why we put the script in `sbin` instead of `bin`.

This means there's no need to specify `sudo` on every flipping command in the script.

I have therefore removed the `sudo` commands from the template.

----

I've removed the `set +e` from the end of the script. The `-e` option is reset to its previous value when the script exits. (This is _not_ the case for scripts you `source`.)

**Review by satish153 — APPROVED** [review]
This is nice. Tested it for es2k and dpdk targets.

**Comment by n-sandeep** [comment]
> `Jinja2`

So, we need to update documents/requirements file to tell user that he needs to install "Jinja2" before building networking-recipe?
If Yes, this package need to be added in ACC python packages list as well.

**Comment by ffoulkes** [comment]
> > `Jinja2`
> 
> So, we need to update documents/requirements file to tell user that he needs to install "Jinja2" before building networking-recipe? If Yes, this package need to be added in ACC python packages list as well.

That's correct.

Do we have a requirements document or standard requirements file? Where do I find it/them?

**Comment by n-sandeep** [comment]
> > > `Jinja2`
> > 
> > 
> > So, we need to update documents/requirements file to tell user that he needs to install "Jinja2" before building networking-recipe? If Yes, this package need to be added in ACC python packages list as well.
> 
> That's correct.
> 
> Do we have a requirements document or standard requirements file? Where do I find it/them?

We have a requirements.txt file under networking-recipe, but looks like not updated at all. Better to start updating that file for all python pakages I believe.

**Review by nupuruttarwar — COMMENTED** [review]
Logic LGTM. Please update requirements.txt to include jinja2. Also, is the PR to include jinja2 in list of ACC packages already in review?

**Comment by ffoulkes** [comment]
> Logic LGTM. Please update requirements.txt to include jinja2. Also, is the PR to include jinja2 in list of ACC packages already in review?

I've added the updated requirements.txt file to the PR.

Jinja2 isn't an ACC requirement. It runs on the development host at build time. I believe there's a Python script to install the build dependencies.

**Review by nupuruttarwar — APPROVED** [review]
LGTM

File: requirements.txt [hunk 1]
```diff
@@ -1 +1,2 @@
 p4runtime==1.3.0
+Jinja2
```
Comment (5abeel) [inline requirements.txt:2]: It is recommended to pin the version down for a repeatable/dependable build outcomes.

Some linters like Docker linters (which this is not), fail when versions are not pinned down

**Pull request merged**